File: src/motion_controller/src/motion_controller__.py
#!/usr/bin/env python

import rospy
from motion_controller.msg import Pwm
from std_msgs.msg import String
from geometry_msgs.msg import Twist
from taotao.msg import Mdu

def callback(data):
	rospy.loginfo(rospy.get_caller_id() + "I heard %s", data)

def talker():
	data = Twist()
	rospy.init_node('motion_controller_py', anonymous=True)
	pub = rospy.Publisher('cmd_vel', Twist, queue_size=10)

	rospy.Subscriber("mdu", Mdu, callback)

	rate = rospy.Rate(1) # 10hz
	data.linear.x = 10;
	while not rospy.is_shutdown():
		hello_str = "hello world %s" % rospy.get_time()
		rospy.loginfo(hello_str)
		pub.publish(data)
		rate.sleep()

if __name__ == '__main__':
    try:
        talker()
    except rospy.ROSInterruptException:
        pass

File: src/motion_controller/src/motion_controller.cpp
#include "ros/ros.h"
#include "std_msgs/String.h"

#include "motion_controller/Pwm.h"
#include "taotao/Mdu.h"

#include <sstream>

void read_data_hoverboard(const taotao::Mdu::ConstPtr&  msg) {
  printf("I heard: [%f]", msg->motor_l.voltage);
}


int main(int argc, char **argv) {
	motion_controller::Pwm msg;
	ros::init(argc, argv, "motion_controller");

	ros::NodeHandle n;
	ros::Publisher pwm_pub = n.advertise<motion_controller::Pwm>("pwm", 100);

	ros::Subscriber pwm_data_sub    = n.subscribe("mdu",100,read_data_hoverboard);

	ros::Rate loop_rate(1);
	while (ros::ok()) {
		msg.pwm_l = 100.0;
		msg.pwm_r = 150.0;

		pwm_pub.publish(msg);

		ros::spinOnce();

		loop_rate.sleep();
  }

  return 0;
}

